components/tidb_query_datatype/src/codec/mysql/enums.rs

```diff
@@ -131,22 +177,114 @@ impl<'a> PartialOrd for EnumRef<'a> {
     }
 }
 
+impl<'a> ToString for EnumRef<'a> {
+    fn to_string(&self) -> String {
+        String::from_utf8_lossy(self.name).to_string()
+    }
+}
+
+pub trait EnumEncoder: NumberEncoder {
+    #[inline]
+    fn write_enum(&mut self, data: EnumRef) -> Result<()> {
+        self.write_u64_le(data.value as u64)?;
+        self.write_bytes(data.name)?;
+        Ok(())
```

Comment:
Do we really need to encode `data.name` ? If this member be transmited by network, there will be performance regression.

Reply:
There are some problems in adjusting TiDB code for this purpose. We can leave it for further optimizations.